infrastructure/persistence/inventory/inventory_repository.go

```diff
@@ -89,8 +89,16 @@ func (c *InventoryClient) GetInventoryClients(ctx context.Context, page int) (In
 	}
 	defer res.Body.Close()
 
-	err = json.NewDecoder(res.Body).Decode(&results)
-	if err != nil {
+	if res.StatusCode != http.StatusOK {
```

Comment:
This is mostly stylistic, but I like to use a `switch` statement on the response status code instead of an `if` statement. It lets me clump the HTTP status codes together based on their ranges, like this:

```go
        switch {
        case statusCode >= 400:
                logger.Error().Msg("sent HTTP response")
        case statusCode >= 300:
                logger.Info().Msg("sent HTTP response")
        case statusCode >= 200:
                logger.Debug().Msg("sent HTTP response")
        }
```

https://github.com/RedHatInsights/config-manager/blob/83532e2963a4a5b82a9fe285ca6e057140197844/internal/http/render/render.go#L36-L43

Reply:
Done! Thanks for pointing out!